### PR TITLE
Electrode and stun revolver changes

### PR DESCRIPTION
**Pull request by ghost**
- Gave a delay to firing electrodes out of tasers so they cannot be spammed anymore
- Removed stun revolver from research
- Gave stun revolver to HoS, in his locker, with a loyalty pin
- Made stun revolver a steal objective

### CODE REVIEW

**Comment by TheVekter** [comment]
>Remove stun revolver from research

Not a fan of this. It still has to have a pin in it to even work in the first place. Also, why delay electrodes? Are we just trying to replace stunning with disabling now? It just seems really unnecessary.

**Comment by paprka** [comment]
The "I don't play ss13" Vekter

**Comment by Cheridan** [comment]
>Are we just trying to replace stunning with disabling now? 

Uh
Basically, yeah.

**Comment by TheVekter** [comment]
@paprka ayy lmao

@Cheridan this feels like a solid nerf to sec, but I don't know what the delay on firing is so idk. Removing stun revolver seems odd because it removes R&D's only option for weapons without Mining getting involved. Also, any consideration for a nerf instead of just removing? Might be better to give it the delay and reduce number of shots to ~3-4.

**Comment by Thunder12345** [comment]
Yeah, improve, don't remove. Nerf the stunvolver if that's what you think it needs, but give the HoS something new and unique for a gun, not an existing gun repackaged as something different.

**Comment by paprka** [comment]
Stop saying improve don't remove you don't know what it means.

**Comment by Thunder12345** [comment]
Once again Paprika realises he's losing and falls back to his good old ad hominems

**Comment by paprka** [comment]
Losing fucking what?

You're the one turning this into some kind of battle, not me. Shitpost on your own time dumbshit.

**Comment by Thunder12345** [comment]
http://i.imgur.com/HExWSek.png
You're the one who stealthedits comments in an attempt to make me look like an idiot

**Comment by paprka** [comment]
>stealthedits

holy fuck you are stupid, i didn't even see your comment when i made that edit

**Comment by dannno** [comment]
You're both retards who get pissy and try to make everything personal and hostile
shut the fuck up and get back on the topic of the PR you stupid cunts

**Comment by Thunder12345** [comment]
So are you actually going to say where the description used to be "Pride of the head of security"?

**Comment by paprka** [comment]
Prize, sorry.

Look, one of our downstreams still has that desc that I PERSONALLY CHANGED on our repo.

https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/modules/research/designs.dm#L1972

Christ, just fuck off already thunder

**Comment by Thunder12345** [comment]
Wow, a downstream has something different, if you could be bothered to dig that up, why not find the actual commit?
P.S. that description is, AFAIK, never even seen in game

**Comment by paprka** [comment]
That's not the point, if you bothered to pay attention to anything ever. That was the intent behind them being in research in the first place, security upgrades from their standard tasers with a cool fire sound. That's it. Traitor balance is irrelevant, they still need firing pins now.

**Comment by Cheridan** [comment]
Unlocked. Don't start fighting again.

**Comment by paprka** [comment]
I don't find the stun revolver removal from rnd such a bad thing because it seems more like a gimmicky sprite-swapped taser that fits the HoS better than some really hi-tech weapon like the literal nuclear reactor energy gun. Research should be focused on making cool, inventive things, not 'a taser with more shots'.

The issue isn't that the stun revolver is to powerful or it's OP or whatever, that doesn't matter to me, it just doesn't seem like a good item for research when it was initially supposed to be an upgraded taser. It's worse than tasers in some respects now, so I'd rather give it to the HoS as his sidearm alternative to the egun he spawns with.

**Comment by Thunder12345** [comment]
Should probably give research a new toy to make up for the stunvolver. Wabbajack gun anyone?

**Comment by paprka** [comment]
I don't get why that is some responsibility on me to do, it's not like a critical part of r&d to print off stun revolvers

**Comment by Thunder12345** [comment]
I never said it was your job to do it, just that someone should give them something new

**Comment by paprka** [comment]
Can you keep on topic please

**Comment by Lumipharon** [comment]
It is more useful in 90% of cases then a taser, since disablers are simply niche use - most of that 10% is having a torch stuck to the taser.

Removing the stun revolver just makes it harder for the crew to get access to more non lethal weapons, since it's the only one that can be created on station (syringe guns don't count).

Also with putting the loyalty pin in the HoS's gun, is the gun itself emaggable or what?

**Comment by paprka** [comment]
No, but if you're going to steal it, you probably have an emag, so getting a loyalty implant from the armory should be easy. If you don't want to risk it for your 10 stuns, then keep it in your satchel!

Also, why do non-security personnel need non-lethal weapons? I think what you meant to say is 'makes it harder for antags to get stuns', which is intended. I have very rarely seen stun revolvers passed out during nuke ops/etc, and it's even harder with the new pin system. Their inclusion in the protolathe is completely irrelevant now.

**Comment by Lumipharon** [comment]
Stun revolvers get (or were) given out all the time though.
Pins made it much rarer, since it just takes way more fucking effort from everyone involved since you have to go between science and armory to put pins on shit, instead of just opening a box. Anyway, very often people (science people, heads, whatever) would ask or be given stun revolvers, for non lethal defence.

Didn't happen so much in nuke, but in other mode you still saw it pretty often.

Either way, firing pins fucked everything up, because it requires far more EFFORT for non antags to get R&D guns, because people are lazy, and for some reason lockboxes had to be removed.

**Comment by Steelpoint** [comment]
This seems like a less effort alternative to my HoH gun PR.

**Comment by Fox-McCloud** [comment]
Injecting yourself with a loyalty implant as an antag so you can use his gun is pretty much a self-defeating action though; you're effectively making yourself a non-traitor by doing that.

Not sure I like the idea of the stun revolver being removed from R&D, but that's me.

**Comment by paprka** [comment]
That's the point steelpoint. The HoS doesn't need some wildly huge fucking 3 mode gun with all these fanzy gizmos, he already has a gun sorta unique to him and this buffs his stunning capaciy (something you complained about)

**Comment by GunHog** [comment]
Loyalty implanting yourself has no gameplay or rules based effect on you as a traitor, so implanting yourself has no effect in that case. I do not know where this nonsense about them being rarely used by the crew is originating, for when I am playing as RD or Scientist, it is the most requested weapon by my Science department Sec officer and the Detective. Sec otherwise does not request weapons until there is a pressing threat (so about 2/5 rounds). In blob/Malf rounds, the HoS will hand out firing pins like CANDY.

**Comment by Alek2ander** [comment]
Tazers have 5 shots. If someone wants to spam and waste them all, why the hell not?

**Comment by Steelpoint** [comment]
The HoS has never had a unique gun to him, which is the point of my PR.

I did a new pass and lowered the values to something more acceptable, I don't have much to say about the other parts of this PR, but I think its redundant to give the HoS a stun revolver when my PR already does that but better.

**Comment by dannno** [comment]
>already does that but better

no it doesn't

**Comment by dannno** [comment]
>How about the HoS gets an engraved stun revolver? HoS gets a distinct gun, R&D keeps their cool gun, everybody wins.

This is a good suggestion and I have sprites you can use if you want to do this

**Comment by paprka** [comment]
I have yet to see a reason why RnD needs to keep their 10 shot taser though, it heavily imbalances traitors who play rnd and always has

**Comment by dannno** [comment]
make it a disable-revolver instead then

**Comment by paprka** [comment]
No, I'm not changing it, I'm removing it because I'm making balance changes here.

The protolathe is still a gun printer. Cheridan's pin's PR didn't alleviate that at all. The shitty 'lel 10 stuns' gun still exists, and you have to understand that this clashes with my design philosophy behind the availability of ranged stuns. It's not my fucking burden to add good new creative things, either. I'm making a change for the sake of balance, and instead of straight up removing the stun revolver, I'm giving it to the HoS so steelpoint's epic wall killer overpowered HoS gun PR can get closed in the process since it's a decidedly terrible PR at this point.

**Comment by Steelpoint** [comment]
Make up your mind pap, the wall breaching fire mode was your suggestion and you told me to do it.

Keep the Stun Revolver as a Disabler Revolver if you must, but leave the HoS gun stuff to me.

**Comment by Fox-McCloud** [comment]
Have to agree with Steelpoint here--especially considering you completely talked up the virtues of a breach shot and how HoS shouldn't be stopped by walls or small actions that antags pull that could pretty much grind him to a halt; heck you even defended it against silicon players who were pooh-poohing it.

**Comment by paprka** [comment]
we nerfed the silicons by taking away their tase and lase, so keeping a feature of the hos that allows him to walk through any wall in the game easily is not a good idea for the balance.

i said one or the other.

you agreed the secborg disabler pr was a better idea.

that one got merged.

now stop trying to peddle your hos gun pr, the sprites are bad and it's snowflake "i want to make the hos super op because it's the job i play the most" bullcrap and incredibly transparent. my pr doesn't mess with the balance of AI antags. your gun pr hasn't been accepted because it's a bad idea now that syndicate borgs have been so heavily nerfed.

**Comment by Steelpoint** [comment]
I have consistently been changing the gun based on feedback, aside from the breach setting the gun is pretty well balanced imo and even then the breaching is a wild card. All I'm saying is that I've already got a PR up that is entirely dedicated to a HoS gun, thus I think any HoS gun work should be kept there.

**Comment by paprka** [comment]
I suggested in your PR that you give the HoS a stun revolver since that allows his stunning capacity (something you complained about) to be massively buffed so he's not a pushover and it also has no delay like electrodes normally do in this PR but also while making it a unique item by removing it from the protolathe.

You're overthinking it, a lot. The same shit happened with the ERT armory. Simplicity and minimalism is good for our codebase and this multi faceted gun is silly. A taser with double capacity is easy to wrap your head around and is unique enough for the HoS without being wickedly OP.

**Comment by Lumipharon** [comment]
The stun revolver used to only have like 5-6 stun shots. A certain someone buffed it at some point.

**Comment by paprka** [comment]
Okay I left the stun revolver in the protolathe, it's a disabler with extra capacity now since traitor scientists REALLY NEED FREE GUNS LOL!!!

I gave the HoS' taser pistol dannos' luger sprites to make it unique as well.

**Comment by paprka** [comment]
@dannno why the fuck didn't you provide the sprites in .dmi or at least 32x32 you moron

**Comment by paprka** [comment]
- Added unique sprites for the HoS' stun pistol, dannos' luger sprites
- Made the stun revolver a disabler with extra capacity, kept it in protolathe
- Raised the combat requirement for the advanced egun

**Comment by dannno** [comment]
Dude. I have a DMI ready. I just didn't know if you wanted to use it or not.
https://www.dropbox.com/s/n86priazav388ns/hosgun.dmi?dl=0

**Comment by paprka** [comment]
It's fine

**Comment by Jordie0608** [comment]
Changelog

**Comment by paprka** [comment]
Changelog! Also, conflicts fixed.

File: code/modules/projectiles/guns/energy/stun.dm [hunk 1]
```diff
@@ -6,12 +6,19 @@
...
+/obj/item/weapon/gun/energy/hospistol
+	name = "stun pistol"
+	desc = "A high-tech handgun that fires reusable stun cartridges in an internal magazine. The prize of the Head of Security."
+	icon_state = "hospistol"
+	ammo_type = list(/obj/item/ammo_casing/energy/electrode/gun)
+	can_flashlight = 0
+	pin = /obj/item/device/firing_pin/implant/loyalty
+
 /obj/item/weapon/gun/energy/stunrevolver
 	name = "stun revolver"
```
Comment (GunHog) [inline code/modules/projectiles/guns/energy/stun.dm]: If you are going through with this nerf, at least change the name to reflect that. It is not a stun revolver any more, and I assume you changed the meter from yellow to cyan as well?

**Comment by Steelpoint** [comment]
Still think we would be better off keeping any HoS Gun related changes to the PR that is dedicated to doing that, I also somewhat disagree with the Loyalty Pin.

File: code/modules/projectiles/guns/energy/stun.dm (hunk 1 above)
Comment (paprka) [inline code/modules/projectiles/guns/energy/stun.dm]: Yes and disablers still stun just less frequently and 'disable revolver' sounds extremely retarded

**Comment by Fox-McCloud** [comment]
Doesn't this essentially mean a massive 10 second delay between taser shots? I can understand 5, but 10 is pretty ridiculous.

**Comment by paprka** [comment]
It's one second. The value is in deciseconds. You can still shoot them pretty fast, but you can't clickspam. Count seconds on your fingers or listen to a clock, that's about how fast you can shoot electrodes now versus the 2 or so per second which is the current rate.

**Comment by Fox-McCloud** [comment]
Ah. My mistake! Thought spawn was in second as opposed to deciseconds, for some reason.

File: code/modules/projectiles/guns/energy/stun.dm [hunk 2]
```diff
@@ -21,6 +28,7 @@
 	icon_state = "advtaser"
 	ammo_type = list(/obj/item/ammo_casing/energy/electrode, /obj/item/ammo_casing/energy/disabler)
 	origin_tech = null
+	fire_delay = 5
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/stun.dm]: Shouldn't this be 10 and not 5? Essentially keeping it on taser, from the start of the round, would be advantageous to shave off that 0.5 a second delay as opposed to switching it from stun to disable then back to stun.

Comment (paprka) [inline code/modules/projectiles/guns/energy/stun.dm]: I have no clue why this is here actually

**Comment by Fox-McCloud** [comment]
Might I recommend upping the time from 10 to 25. In this case, if you miss, then you could have gotten off 5 disablers shot (1.5 stuns, effectively), which gives a far more serious incentive to use the disable setting if you have poor aim or don't have a guaranteed line-up.

**Comment by paprka** [comment]
- Removed HoS pistol
- Stun revolvers use electrodes again, but have a delay like normal tasers (but double the capacity)
- Advanced energy guns no longer use electrodes, they are laser/disabler like regular e-guns
- Shifted some weapons to the front of lockers

**Pull request merged**